src/UnitGenerator/SourceGenerator.cs

```diff
@@ -544,34 +778,183 @@ public int CompareTo({{unitTypeName}} other)
+        public static bool IsFinite({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsFinite(value.value);
+        public static bool IsImaginaryNumber({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsImaginaryNumber(value.value);
+        public static bool IsInfinity({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsInfinity(value.value);
+        public static bool IsInteger({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsInteger(value.value);
+        public static bool IsNaN({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsNaN(value.value);
+        public static bool IsNegative({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsNegative(value.value);
+        public static bool IsNegativeInfinity({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsNegativeInfinity(value.value);
+        public static bool IsNormal({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsNormal(value.value);
+        public static bool IsOddInteger({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsOddInteger(value.value);
+        public static bool IsPositive({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsPositive(value.value);
+        public static bool IsPositiveInfinity({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsPositiveInfinity(value.value);
+        public static bool IsRealNumber({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsRealNumber(value.value);
+        public static bool IsSubnormal({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsSubnormal(value.value);
+        public static bool IsZero({{unitTypeName}} value) => global::UnitGenerator.AsNumber<{{innerTypeName}}>.IsZero(value.value);
+        public static {{unitTypeName}} MaxMagnitude({{unitTypeName}} x, {{unitTypeName}} y) => new(global::UnitGenerator.AsNumber<{{innerTypeName}}>.MaxMagnitude(x.value, y.value));
+        public static {{unitTypeName}} MaxMagnitudeNumber({{unitTypeName}} x, {{unitTypeName}} y) => new(global::UnitGenerator.AsNumber<{{innerTypeName}}>.MaxMagnitudeNumber(x.value, y.value));
+        public static {{unitTypeName}} MinMagnitude({{unitTypeName}} x, {{unitTypeName}} y) => new(global::UnitGenerator.AsNumber<{{innerTypeName}}>.MinMagnitude(x.value, y.value));
+        public static {{unitTypeName}} MinMagnitudeNumber({{unitTypeName}} x, {{unitTypeName}} y) => new(global::UnitGenerator.AsNumber<{{innerTypeName}}>.MinMagnitudeNumber(x.value, y.value));
+
+        public static bool TryConvertFromChecked<TOther>(TOther value, out {{unitTypeName}} result) where TOther : INumberBase<TOther> 
+        {
+            throw new NotSupportedException();
```

Comment:
The methods of`TryConvert*` family are in `INumberBase<T>` but cannot be called because they are protected. I compromised and made it NotSupported....
Please let me know if you have a better way.

Reply:
Fixed implementation to `return false`.